### PR TITLE
Bug 1934400: scc-admission: don't apply defaultAllowPrivilegeEscalation:false when container is privileged

### PR DESCRIPTION
**Pull request by sttts**
This is a proposal.

It is the counterpart of the following from the pod validation in upstream:

```golang
    if sc.AllowPrivilegeEscalation != nil && !*sc.AllowPrivilegeEscalation {
        if sc.Privileged != nil && *sc.Privileged {
            allErrs = append(allErrs, field.Invalid(fldPath, sc, "cannot set `allowPrivilegeEscalation` to false and `privileged` to true"))
        }

        if sc.Capabilities != nil {
            for _, cap := range sc.Capabilities.Add {
                if string(cap) == "CAP_SYS_ADMIN" {
                    allErrs = append(allErrs, field.Invalid(fldPath, sc, "cannot set `allowPrivilegeEscalation` to false and `capabilities.Add` CAP_SYS_ADMIN"))
                }
            }
        }
    }
```

### CODE REVIEW

**Comment by sttts** [comment]
/hold

until we agree to do this.

File: pkg/securitycontextconstraints/sccmatching/provider.go [hunk 1]
```diff
@@ -191,9 +191,20 @@ func (s *simpleProvider) CreateContainerSecurityContext(pod *api.Pod, container
 		sc.SetReadOnlyRootFilesystem(&readOnlyRootFS)
 	}
 
+	isPrivileged := sc.Privileged() != nil && *sc.Privileged()
```
Comment (p0lyn0mial) [inline pkg/securitycontextconstraints/sccmatching/provider.go:194]: do you know why `sc.Privileged()` doesn't take into account capabilities?

Comment (p0lyn0mial) [reply pkg/securitycontextconstraints/sccmatching/provider.go:194]: ah, okay this is just a flag, right?

Comment (sttts) [reply pkg/securitycontextconstraints/sccmatching/provider.go:194]: it's only an accessor for the privileged field. Maybe my local var is misnamed.

File: pkg/securitycontextconstraints/sccmatching/provider.go [hunk 2]
```diff
@@ -191,9 +191,20 @@ func (s *simpleProvider) CreateContainerSecurityContext(pod *api.Pod, container
...
+	if caps != nil {
+		for _, cap := range caps.Add {
+			if string(cap) == "CAP_SYS_ADMIN" {
+				isPrivileged = true
```
Comment (stlaz) [inline pkg/securitycontextconstraints/sccmatching/provider.go:198]: break

File: pkg/securitycontextconstraints/sccmatching/provider.go [hunk 3]
```diff
@@ -191,9 +191,20 @@ func (s *simpleProvider) CreateContainerSecurityContext(pod *api.Pod, container
 		sc.SetReadOnlyRootFilesystem(&readOnlyRootFS)
 	}
 
+	isPrivileged := sc.Privileged() != nil && *sc.Privileged()
+	if caps != nil {
```
Comment (stlaz) [inline pkg/securitycontextconstraints/sccmatching/provider.go:195]: if !isPrivileged && caps != nil

File: pkg/securitycontextconstraints/sccmatching/provider.go (hunk 1 above)
Comment (stlaz) [reply pkg/securitycontextconstraints/sccmatching/provider.go:194]: `isHighlyPrivileged` or `isRoot` maybe?

**Review by stlaz — CHANGES_REQUESTED** [review]

File: pkg/securitycontextconstraints/sccmatching/provider.go (hunk 1 above)
Comment (sttts) [reply pkg/securitycontextconstraints/sccmatching/provider.go:194]: `isPrivileged` and `addCapSysAdmin`

File: pkg/securitycontextconstraints/sccmatching/provider.go (hunk 2 above)
Comment (sttts) [reply pkg/securitycontextconstraints/sccmatching/provider.go:198]: fixed

File: pkg/securitycontextconstraints/sccmatching/provider.go (hunk 3 above)
Comment (sttts) [reply pkg/securitycontextconstraints/sccmatching/provider.go:195]: obsolete

File: pkg/securitycontextconstraints/sccmatching/provider.go (hunk 2 above)
Comment (stlaz) [reply pkg/securitycontextconstraints/sccmatching/provider.go:199]: `addCapSysAdmin = true`

**Pull request merged**